spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configuration-watcher/src/main/java/org/springframework/cloud/kubernetes/configuration/watcher/ConfigurationWatcherAutoConfiguration.java

```diff
@@ -53,31 +54,30 @@ public WebClient webClient(WebClient.Builder webClientBuilder) {
 
 	@Bean
 	@ConditionalOnMissingBean(ConfigMapWatcherChangeDetector.class)
-	@ConditionalOnBean(Fabric8ConfigMapPropertySourceLocator.class)
+	@ConditionalOnBean(KubernetesClientConfigMapPropertySourceLocator.class)
 	public ConfigMapWatcherChangeDetector httpBasedConfigMapWatchChangeDetector(AbstractEnvironment environment,
-			KubernetesClient kubernetesClient,
-			Fabric8ConfigMapPropertySourceLocator fabric8ConfigMapPropertySourceLocator,
-			Fabric8SecretsPropertySourceLocator fabric8SecretsPropertySourceLocator, ConfigReloadProperties properties,
-			ConfigurationUpdateStrategy strategy,
+			CoreV1Api coreV1Api, KubernetesClientConfigMapPropertySourceLocator configMapPropertySourceLocator,
+			ConfigReloadProperties properties, ConfigurationUpdateStrategy strategy,
 			ConfigurationWatcherConfigurationProperties k8SConfigurationProperties,
-			ThreadPoolTaskExecutor threadFactory, WebClient webClient,
-			KubernetesReactiveDiscoveryClient kubernetesReactiveDiscoveryClient) {
-		return new HttpBasedConfigMapWatchChangeDetector(environment, properties, kubernetesClient, strategy,
-				fabric8ConfigMapPropertySourceLocator, k8SConfigurationProperties, threadFactory, webClient,
+			KubernetesNamespaceProvider namespaceProvider, ThreadPoolTaskExecutor threadFactory, WebClient webClient,
+			KubernetesInformerReactiveDiscoveryClient kubernetesReactiveDiscoveryClient) {
+		return new HttpBasedConfigMapWatchChangeDetector(coreV1Api, environment, properties, strategy,
+				configMapPropertySourceLocator, namespaceProvider, k8SConfigurationProperties, threadFactory, webClient,
 				kubernetesReactiveDiscoveryClient);
 	}
 
 	@Bean
 	@ConditionalOnMissingBean(SecretsWatcherChangeDetector.class)
```

Comment:
**[UNRELATED]** Nit, but the class is redundant in this case. This is consistent across this auto-config and maybe this is intentional to be more explicit?

Reply:
Are you talking about the `@ConditionalOnMissingBean`?

Reply:
Yeh, I could have been more explicit in my comment. When type is not specified it goes off the return type of the method.

Reply:
Got you, yes!